src/task2/flattening_module/imageflattener.py

```diff
@@ -185,52 +227,64 @@ def done():
                 print('That\'s not a number.')
                 return
 
-            self.__dim = (short, long)
-
+            if self.__selected_panel is None:
+                print("No panel is selected")
+                pop.destroy()
+            self.__selected_panel.dims = (short, long)
+            self.__last_dims = (short, long)
             pop.destroy()
             self._flatten_img()
 
         done_button = Button(pop, text='Done', command=done)
         done_button.pack(side=LEFT)
 
-        if self.dim is not None:
+        if self.__last_dims is not None:
             skip_button = Button(pop, text='Use Last',
-                                 command=skip)
+                                 command=use_last)
             skip_button.pack(side=LEFT)
 
         pop.mainloop()
 
 
-def _parse_input(images_input: any) -> Dict[str, List[Tuple[int, int]]]:
+def parse_input(images_input: any) -> Dict[str, List[Panel]]:
     """Attempts to parse input from either list or dictionary if input is
     bad will throw useful errors"""
-
     panels_in = dict()
-
-    if isinstance(images_input, dict):
-        for key in images_input:
-            if isinstance(images_input[key], list):
-                panels_in[key] = []
-                for point in images_input[key]:
-                    if isinstance(point, list) \
-                       and len(images_input[key]) == 2 \
-                       and isinstance(point[0], int) \
-                       and isinstance(point[1], int):
-
-                        panels_in[key] = tuple(images_input[key])
-
-                    else:
-
-                        raise JsonFormatError('lists should contain lists of ints\
-                                              of len 2')
+    if not isinstance(images_input, list):
+        raise JsonFormatError('Json should be formatted like a list')
+
+    for image_listing in images_input:
+        if ('file' not in image_listing or
+            'gps' not in image_listing or
+                'pixels' not in image_listing):
+            raise JsonFormatError('All items must contain a \"file\", \
+                                  \"gps\" and \"pixle\" keys.')
+
+        file = image_listing['file']
+        gpss = image_listing['gps']
+        pixels = image_listing['pixels']
+
+        print(file, gpss, pixels)
+
+        if not isinstance(file, str):
+            raise JsonFormatError('File must be string')
+        if not isinstance(gpss, list) or not isinstance(pixels, list) or \
```

Comment:
Hey at least your code will not break tests if we decide to add a PEP8 merge enforcer into the repo xD